File: Tracks/typescript/raindrops/raindrops.ts
type RaindropMapper = { factor: number; text: string };

export default class Raindrops {
    private readonly defaultMapper: RaindropMapper[] = [
        { factor: 3, text: "Pling" },
        { factor: 5, text: "Plang" },
        { factor: 7, text: "Plong" }
    ];
    private readonly mapper: RaindropMapper[];
    constructor(mapper?: RaindropMapper[]) {
        this.mapper = mapper || this.defaultMapper;
    }
    convert = (input: number): string =>
        this.mapper.reduce((result, { factor, text }) => result + (input % factor === 0 ? text : ""), "") ||
        input.toString();
}

File: Tracks/csharp/raindrops/Raindrops.cs
using System.Collections.Generic;
using System.Linq;

public static class Raindrops
{
    private static readonly SortedDictionary<int, string> plinger = new SortedDictionary<int, string> { { 3, "Pling" }, { 5, "Plang" }, { 7, "Plong" } };
    public static string Convert(int number) => string.Join("", plinger.Where((keyPair) => number % keyPair.Key == 0)
                                                                       .Select((keyPair) => keyPair.Value)
                                                                       .DefaultIfEmpty(number.ToString()));
}